phlib/util.c

```diff
@@ -161,6 +161,226 @@ VOID PhCenterWindow(
     }
 }
 
+LONG PhGetDpi(
+    _In_ LONG Number,
+    _In_ LONG DpiValue
+    )
+{
+    return PhMultiplyDivideSigned (Number, DpiValue, USER_DEFAULT_SCREEN_DPI);
+}
+
+LONG PhGetMonitorDpi(
+    _In_ LPCRECT rect
+    )
+{
+    return PhGetDpiValue(NULL, rect);
+}
+
+LONG PhGetTaskbarDpi(
+    VOID
+    )
+{
+    APPBARDATA taskbar_rect = {0};
+
+    taskbar_rect.cbSize = sizeof (APPBARDATA);
+
+    if (SHAppBarMessage (ABM_GETTASKBARPOS, &taskbar_rect))
+        return PhGetMonitorDpi (&taskbar_rect.rc);
+
+    return PhGetDpiValue(NULL, NULL);
+}
+
+LONG PhGetWindowDpi(
+    _In_ HWND hwnd
+    )
+{
+    RECT rect;
+
+    if (!GetWindowRect (hwnd, &rect))
+        return PhGetDpiValue(hwnd, NULL);
+
+    return PhGetDpiValue(NULL, &rect);
+}
+
+LONG PhGetDpiValue(
+    _In_opt_ HWND hwnd,
+    _In_opt_ LPCRECT rect
+    )
+{
+    static PH_INITONCE initOnce = PH_INITONCE_INIT;
+    static GDFM _GetDpiForMonitor = NULL; // win81+
+    static GDFW _GetDpiForWindow = NULL; // win10rs1+
+    static GDFS _GetDpiForSystem = NULL; // win10rs1+
+
+    HMONITOR hmonitor;
+    HINSTANCE hshcore;
+    HINSTANCE huser32;
```

Comment:
HINSTANCE type is deprecated and should be PVOID

Reply:
did not see this in HACKING.md